dask/dataframe/io/demo.py

```diff
@@ -128,8 +337,9 @@ def make_timeseries_part(start, end, dtypes, columns, freq, state_data, kwargs):
         if k in columns:
             data[k] = result
     df = pd.DataFrame(data, index=index, columns=columns)
-    if df.index[-1] == end:
-        df = df.iloc[:-1]
+    for k, dtype in dtypes.items():
+        if k in columns:
+            df[k] = df[k].astype(dtype)
```

Comment:
@phofl I have a feeling that this line might be creating problems in terms of performance. Other changes in this PR don't look like they could be significant. What do you think?

Reply:
[`astype`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.astype.html) accepts a mapping and is much more performant when operating on the entire dataframe than column wise.
We may also want to set `copy=False`

Reply:
Yep that's correct, but the magnitude surprises me.

Reply:
I'll see if I can avoid this `astype` entirely.